bin/init

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# This script will ensure Portus' database is ready to be used. It will keep
+# waiting for the db to be usable, but the script will exit with an error
+# after a certain amount of failed attempts.
+#
+# The script will automatically import all the SSL certificates from
+# `/certificates` into the final system. This is needed to talk with the
+# registry API when this one is protected by TLS.
+#
+# Finally the script will start apache running Portus via mod_rails.
+
+set -e
+# check that BUNDLER_BIN is set
+if [ ! -z $BUNDLER_BIN ]; then
+    echo "BUNDLER_BIN var is not set. Please set that and try again" >&2
+    exit -1
+fi
+
+setup_database() {
+  set +e
+
+  TIMEOUT=90
+  COUNT=0
+  RETRY=1
+
+  export SKIP_MIGRATION=1
+
+  while [ $RETRY -ne 0 ]; do
+    case $(SKIP_MIGRATION=1 $BUNDLER_BIN exec rails r /check_db.rb | grep DB) in
+      "DB_DOWN")
+        if [ "$COUNT" -ge "$TIMEOUT" ]; then
+          printf " [FAIL]\n"
+          echo "Timeout reached, exiting with error"
+          exit 1
+        fi
+        echo "Waiting for mariadb to be ready in 5 seconds"
+        sleep 5
+        COUNT=$((COUNT+5))
+        ;;
+      "DB_EMPTY"|"DB_MISSING")
+        # create db, apply schema and seed
+        echo "Initializing database"
+        SKIP_MIGRATION=1 $BUNDLER_BIN exec rake db:create
+        SKIP_MIGRATION=1 $BUNDLER_BIN exec rake db:schema:load
+        if [ $? -ne 0 ]; then
+            echo "Error at setup time"
+            exit 1
+        fi
+        ;;
+      "DB_READY")
+        echo "Database ready"
+        break
+        ;;
+    esac
+  done
+  set -e
+}
+
+cd /srv/velum
+setup_database
+$BUNDLER_BIN exec rake db:migrate
+$BUNDLER_BIN exec "rails s -b 0.0.0.0 -p 3000 Puma"
```

Comment:
Here I'd use `rails s -b 0.0.0.0 -p $PORT Puma` and let's set this `PORT` envvar in our manifests once and for all...